### PR TITLE
#111 single quoted strings

### PR DESCRIPTION
**Pull request by abynenkov-ib**
plus null pointer exception fix

### CODE REVIEW

File: pkg/lexer/lexer.go [hunk 1]
```diff
@@ -107,9 +107,10 @@ func (l *Lexer) readNumber() string {
 }
 
 func (l *Lexer) readLiteral() string {
+	startQuota := l.ch // it's a start string quota
```
Comment (rchowinfoblox) [inline pkg/lexer/lexer.go:110]: i believe the word u want to use is "quote" instead of "quota", as in "startQuote"

File: pkg/lexer/lexer.go [hunk 2]
```diff
@@ -132,8 +133,8 @@ func isIdentifierChar(ch byte) bool {
 	return isLetter(ch) || ch == '.' || ch == '*' || ch == '@' || ch == '[' || ch == ']' || ch == '"'
```
Comment (rchowinfoblox) [inline pkg/lexer/lexer.go:133]: do we also need to add `ch == '\''`

**Comment by rchowinfoblox** [comment]
Not sure if we're supposed to allow single-quoted strings in SEAL, I'll let others comment on that.

**Review by rchowinfoblox — APPROVED** [review]
LGTM

File: pkg/ast/ast.go [hunk 3]
```diff
@@ -221,7 +221,9 @@ func (slf *PrefixCondition) String() string {
 }
 func (slf *PrefixCondition) GetTypes() []*Identifier {
 	out := []*Identifier{}
-	out = append(out, slf.Right.GetTypes()...)
+	if slf.Right != nil {
```
Comment (wplu) [inline pkg/ast/ast.go:224]: any comparison to nil needs to use IsNilInterface()
```suggestion
	if !types.IsNilInterface(slf.Right) {
```

need to use this same pattern when comparing interfaces

**Comment by wplu** [comment]
we discussed this and do not see a need for it.   every feature requires maintenance.  it is harder to remove something once we support it - we only want to implement this when needed.

**Comment by rchowinfoblox** [comment]
If we don't support single-quoted strings, we should still fix the coredump and instead return an error when a single-quoted string is attempted as indicated in issue #111

**Comment by wplu** [comment]
i agree - @abynenkov-ib please return appropriate error

**Pull request closed without merging**